packages/calendar/src/compute/timeRange.ts

```diff
@@ -184,12 +184,8 @@ const getTimeInterval = (firstWeekday: Weekday) => {
 function shiftArray<T>(arr: T[], x: number): T[] {
     if (!arr.length || !x) return arr
 
-    for (let i = 0; i < x; i++) {
-        const shifted = arr.shift() as T
-        arr.push(shifted)
-    }
-
-    return arr
+    x = x % arr.length
+    return arr.slice(x, arr.length).concat(arr.slice(0, x))
```

Comment:
Alternatively, you can create a new array and build it based on the shifted elements from the original array.

```ts
function shiftArray<T>(arr: T[], x: number): T[] {
    if (!arr.length || !x) return arr;

    const newArr = [...arr];

    for (let i = 0; i < x; i++) {
        const shifted = newArr.shift() as T;
        newArr.push(shifted);
    }

    return newArr;
}
```

Reply:
However, I liked your fix 🙏